harness/determined/pytorch/_data.py

```diff
@@ -191,6 +203,18 @@ def get_data_loader(
             worker_init_fn=self.worker_init_fn,  # type: ignore
         )
 
+    def __iter__(self) -> Iterator:
+        """Compatibiliy with the real DataLoader when using a PyTorchTrial outside of Determined."""
+        return iter(self.get_data_loader())
+
+    def __len__(self) -> int:
+        """Compatibiliy with the real DataLoader when using a PyTorchTrial outside of Determined."""
+        # TODO(DET-1524): uncomment this logic.
```

Comment:
can you be more prescriptive on when we should uncomment this logic?

Reply:
Sure.